libs/api/domains/syslumenn/src/lib/syslumenn.service.ts

```diff
@@ -1,62 +1,157 @@
-import { SyslumennClient } from './client/syslumenn.client'
 import { Homestay, mapHomestay } from './models/homestay'
 import {
   SyslumennAuction,
   mapSyslumennAuction,
 } from './models/syslumennAuction'
-import { Injectable } from '@nestjs/common'
-import {
-  Person,
-  Attachment,
-  DataUploadResponse,
-  SealedCriminalRecordResponse,
-} from './models/dataUpload'
+import { Injectable, Inject } from '@nestjs/common'
+import { constructUploadDataObject } from './models/dataUpload'
+
+import { Attachment, Person } from './dto/uploadData.input'
 import {
   OperatingLicense,
   mapOperatingLicense,
 } from './models/operatingLicense'
+import {
+  SyslumennApi,
+  SyslumennApiConfig,
+  VirkarHeimagistingar,
+  Uppbod,
+  VirkLeyfi,
+  VottordSkeyti,
+  Skilabod,
+  SvarSkeyti,
+  EmbaettiOgStarfsstodvar,
+} from '@island.is/clients/syslumenn'
+import { Auth, AuthMiddleware } from '@island.is/auth-nest-tools'
+import {
+  CertificateInfoRepsonse,
+  mapCertificateInfo,
+} from './models/certificateInfo'
+import {
+  DistrictCommissionersAgenciesRepsonse,
+  mapDistrictCommissionersAgenciesRepsonse,
+} from './models/districtCommissionersAgencies'
 
+const SYSLUMENN_CLIENT_CONFIG = 'SYSLUMENN_CLIENT_CONFIG'
 @Injectable()
 export class SyslumennService {
-  constructor(private syslumennClient: SyslumennClient) {}
+  private id = ''
+  private accessToken = ''
+
+  constructor(
+    private syslumennApi: SyslumennApi,
+    @Inject(SYSLUMENN_CLIENT_CONFIG)
+    private clientConfig: SyslumennApiConfig,
+  ) {}
+
+  private async login() {
+    const config = {
+      notandi: this.clientConfig.username,
+      lykilord: this.clientConfig.password,
+    }
+
+    const { audkenni, accessToken } = await this.syslumennApi.innskraningPost({
+      notandi: config,
+    })
+    if (audkenni && accessToken) {
+      this.id = audkenni
+      this.accessToken = accessToken
+    }
+  }
+
+  syslumennApiWithAuth() {
+    const auth = {
+      scope: [],
+      authorization: `Bearer ${this.accessToken}`,
+      client: 'client-syslumenn',
+    } as Auth
+
+    return this.syslumennApi.withMiddleware(new AuthMiddleware(auth))
+  }
 
   async getHomestays(year?: number): Promise<Homestay[]> {
-    const homestays = await this.syslumennClient.getHomestays(year)
+    await this.login()
+
+    const homestays = year
+      ? await this.syslumennApi.virkarHeimagistingarGet({
+          audkenni: this.id,
+          ar: year ? JSON.stringify(year) : null,
+        })
+      : await this.syslumennApi.virkarHeimagistingarGetAll({
+          audkenni: this.id,
+        })
 
-    return (homestays ?? []).map(mapHomestay)
+    return ((homestays as VirkarHeimagistingar[]) ?? []).map(mapHomestay)
```

Comment:
this looks iffy, not sure why we need to cast this

Reply:
This was an outdated change, with previous clientConfigs. Will revert